### PR TITLE
Check idris version before building

### PR DESCRIPTION
**Pull request by symbiont-arnaud-bailly**
The actual version might actually not be correct...

### CODE REVIEW

**Comment by clayrat** [comment]
This doesn't seem future-proof, e.g. what if 1.3.3 comes out?

**Comment by symbiont-arnaud-bailly** [comment]
You change the reference? The point is that one needs a specific version of Idris to build Idris2 so better pin it down in the Makefile and ensure someone building Idris2 will get an explicit error if his Idris installation is incorrect. Having a version range and a way to compare them would be better in the long run.

**Comment by symbiont-arnaud-bailly** [comment]
But sure, having a version range and a way to compare them would be better.

**Comment by melted** [comment]
The release builds of Idris, such as the binary builds or the tarballs on hackage doesn't feature the git commit part. It's way too narrow a test anyway, pinning it to a specific commit makes it impossible to work on Idris 1 without hassle.

**Comment by symbiont-arnaud-bailly** [comment]
I totally agree, hence my comment:

> The actual version might actually not be correct...

What's the correct version that should be checked?

**Comment by melted** [comment]
the version would differ depending on whether it was built from git or from hackage (or using a binary build), the git commit part is omitted on release builds. You could just check if it starts with 1.3.2, because we know that all 1.3.2 works.

**Comment by melted** [comment]
Try this 

```
$(expr "$(idris --version)" : '1.3.[2-9].*')
```

**Comment by melted** [comment]
really, that should be `\.` as `.` matches anything. But of course `.` is something  :-)

File: Makefile [hunk 1]
```diff
@@ -1,14 +1,20 @@
 PREFIX ?= ${HOME}/.idris2
+IDRIS_VERSION := $(shell idris --version)
+VALID_IDRIS_VERSION_REGEXP = "1.3.3.*"
```
Comment (melted) [inline Makefile]: 1.3.2

**Comment by edwinb** [comment]
I'd hope that Idris2 would still build with some hypothetical future version of Idris 1, but since there isn't one yet we can worry about that later... also at least in my case, all the changes I've been making to Idris 1 for a while have been to support the development of Idris 2...

Anyway, as it is, this should hopefully help with explaining installation problems!

**Pull request merged**